src/overlays/actors/ovl_En_Kusa/z_en_kusa.h

```diff
@@ -7,13 +7,39 @@ struct EnKusa;
 
 typedef void (*EnKusaActionFunc)(struct EnKusa*, PlayState*);
 
+#define KUSA_GET_TYPE(thisx)((thisx)->params & 0x3)
+
+// `ENKUSA_TYPE_BUSH` and `ENKUSA_TYPE_GRASS`
+#define KUSA_GET_RAND_COLLECTIBLE_ID(thisx) (((thisx)->params >> 8) & 0xF)
+#define KUSA_GET_PARAM_0C(thisx) (((thisx)->params >> 12) & 1)
+
+// `ENKUSA_TYPE_REGROWING_GRASS`, `ENKUSA_TYPE_BUSH` and `ENKUSA_TYPE_GRASS`
+#define KUSA_SHOULD_SPAWN_BUGS(thisx) (((thisx)->params >> 4) & 1)
+
+// `ENKUSA_TYPE_GRASS_2`
+#define KUSA_GET_COLLECTIBLE_ID(thisx) (((thisx)->params >> 8) & 0x7F)
+#define KUSA_GET_PARAM_3F(thisx) (((thisx)->params >> 2) & 0x3F)
+
 typedef enum {
     /* 0 */ ENKUSA_TYPE_BUSH,
     /* 1 */ ENKUSA_TYPE_REGROWING_GRASS,
     /* 2 */ ENKUSA_TYPE_GRASS,
     /* 3 */ ENKUSA_TYPE_GRASS_2
 } EnKusaType;
 
+#define KUSA_BUSH_PARAMS(spawnBugs, randCollectibleId, param0C) \
+    (((ENKUSA_TYPE_BUSH) & 0x3) | (((spawnBugs) & 1) << 4) | (((randCollectibleId) & 0xF) << 8) | (((param0C) & 1) << 12))
```

Comment:
I think it's still valuable to have the mask there, because it shows you can't add a 5th value to the enum without extending past the enum, `& 0x3` shows the types are limited to 0-3 without rearranging the macros. But then it is weird in cases like this where the bitpacking of params will change depending on the type.

Reply:
```suggestion
    ((ENKUSA_TYPE_BUSH & 0x3) | (((spawnBugs) & 1) << 4) | (((randCollectibleId) & 0xF) << 8) | (((param0C) & 1) << 12))
```
Sure that makes sense. I would still argue removing the parens though